cirq-core/cirq/circuits/moment.py

```diff
@@ -135,9 +135,9 @@ def from_ops(cls, *ops: 'cirq.Operation') -> 'cirq.Moment':
     def operations(self) -> Tuple['cirq.Operation', ...]:
         return self._operations
 
-    @property
+    @cached_property
     def qubits(self) -> FrozenSet['cirq.Qid']:
-        return self._qubits
+        return frozenset(self._qubit_to_op)
```

Comment:
I think the value of `qubits()` here is not equivalent to what it was before. In the previous code, `self._qubits` gets set to `self._qubit_to_op.keys()` on line 116, which means that the value returned by `qubits()` was the keys value, not `_qubit_to_op`. By contrast, this new `qubit()` returns `_qubit_to_op` directly, which is a dict.

Could you please double check this?

(Also, it would be good to check that test cases can distinguish between the two kinds of values. If, say, `qubit()` is used by iterating over the values, then it might gloss over the difference.)

Reply:
It's the same. Standard shortcut in Python.

![image](https://github.com/user-attachments/assets/964d88dc-1e3a-41fe-a48a-97d0c748a387)

Reply:
Guess I haven't had occasion to use frozenset enough! Thanks for the tip.